src/Label.php

```diff
@@ -0,0 +1,428 @@
+    // are not added to the label.
+    unset($attributes['Tiefe']);
+    unset($attributes['Breite']);
+    unset($attributes['Höhe']);
+
+    $data['attributes'] = $attributes;
+
+    $qr_code_size = $labelFormat[1] === 'landscape' ? self::QR_CODE_SIZE_LANDSCAPE : self::QR_CODE_SIZE_PORTRAIT;
+    $qr_code = static::getProductQrCode($post_id, $qr_code_size);
+
+    $data['qr_code'] = 'data:image/png;base64,' . base64_encode($qr_code);
+
+    return Pdf::render($data, $labelFormat[0], $labelFormat[1]);
+  }
+
+  /**
+   * Deletes expired files in products price label documents folder.
+   *
+   * This method is run by a CRON scheduled task.
+   */
+  public static function deleteOldPriceLabels() {
+    $fileSystemIterator = new FilesystemIterator(static::$pdfDocsFolderPath);
+    $now = time();
+    foreach ($fileSystemIterator as $file) {
+      if ($now - $file->getCTime() >= static::$pdfDocsExpiryTime) {
+        unlink(static::$pdfDocsFolderPath . '/' . $file->getFilename());
+      }
+    }
+  }
+
+  /**
+   * Adds price label printing controls to a given product.
+   *
+   * @param int $product_id
+   *   The product unique identifier.
+   * @param string $labelFormatDefault
+   *   The default format for the price label.
+   * @param bool $enabled
+   *   If TRUE the controls are enabled.
+   */
+  public static function addProductPriceLabelControls($product_id, $labelFormatDefault, $enabled = TRUE) {
+    Label::displayPriceLabelFormatsSelect(Label::PDF_LABEL_FORMATS, $labelFormatDefault, $enabled);
+    Label::displayPriceLabelPrintButton($product_id, __('Print Sale PDF label', Plugin::L10N), $enabled);
+  }
+
+  /**
+   * Displays a select list with the available price label formats.
+   *
+   * @param array $labelFormats
+   *   Available sizes, orientation and font size for the price label.
+   * @param string $labelFormatDefault
+   *   Default label format.
+   */
+  public static function displayPriceLabelFormatsSelect(array $labelFormats, $labelFormatDefault = '') {
+    if (empty($labelFormatDefault)) {
+      $labelFormatDefault = array_keys($labelFormats)[2];
+    }
+    $selected = ' selected="selected"';
+    ?>
+    <p class="form-field price_label_print">
+      <select
+        class="<?= Plugin::PREFIX ?>-format"
+        style="margin-right: 8px;"
+      >
+      <?php foreach ($labelFormats as $key => $format): ?>
+        <option
+          value="<?= $key ?>"
+          <?= $key === $labelFormatDefault ? $selected : '' ?>>
+          <?= $format ?>
+        </option>
+      <?php endforeach; ?>
+      </select>
+    <?php
+  }
+
+  /**
+   * Displays the button to print the price label.
+   *
+   * @param int $product_id
+   *   Product unique identifier.
+   * @param string $printLabelButtonText
+   *   Text to show in the print label button.
+   */
+  public static function displayPriceLabelPrintButton($product_id, $printLabelButtonText) {
+    ?>
+      <button
```

Comment:
Instead of using the AJAX approach the following should work fine:

- Transform the button into a link and add a query parameter (there might be better approaches for doing so)
  You might need to transform this into a form so you can pass the value of the paper size/orientation as well
  ```
  <a href="<?= $_SERVER['PHP_SELF'] . '?' . $_SERVER['QUERY_STRING'] ?>&action=print" target="_blank"
  ```
- Remove all the JS and PHP AJAX related code as it's unnecessary
- Changes in Pdf.php
  ```diff
  $dompdf->render();
  - return $dompdf->output;
  + $dompdf->stream();
  + die();
  ```

Reply:
FYI: https://github.com/dompdf/dompdf/wiki/Usage#version-070-or-newer